pkg/controller/argocd/service.go

```diff
@@ -468,3 +478,19 @@ func (r *ReconcileArgoCD) reconcileServices(cr *argoprojv1a1.ArgoCD) error {
 	}
 	return nil
 }
+
+func mergeAnnotations(existing, svc *metav1.ObjectMeta) bool {
+	changed := false
+	if existing == nil {
```

Comment:
shouldn't we be checking for `existing.Annotations == nil` rather than the objectMeta? I doubt ObjectMeta would be nil if there's an existing service object!

Reply:
Updated the condition!